packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -215,6 +215,10 @@ export function createInsightsMiddleware<
         });
 
         instantSearchInstance.sendEventToInsights = (event: InsightsEvent) => {
+          // Pass credentials in event payload, in case init was called with different credentials
+          event.payload.appId = event.payload.appId || appId;
+          event.payload.apiKey = event.payload.apiKey || apiKey;
+
```

Comment:
With the latest direction (https://github.com/algolia/search-insights.js/pull/398, will be backported to v2 once it's settled), credentials will probably be added as custom headers through a third argument of `insights()`.

Reply:
that's a little annoying for `onEvent` as it takes only one argument now 🤔

Reply:
A customer-defined onEvent won't need to handle custom credentials, no? Or am I assuming too much?

Reply:
I was just thinking of the default use case of "I'm sending to another service, but still using algolia as well". They'd have to know the second arg of insightsClient too when they're in the "multiple apps" situation too

Reply:
alternatively what we could do is `onEvent(event, Object.assign(insightsClient, credentials), insightsClient)` as that's backwards compatible still

Reply:
I'm mainly seeing this as a possible issue as we document `onEvent(event, aa) { aa(event); ... }`, which would be forced to be different, causing people to possibly lose credentials. That might not be too much of an issue when we still init, but I'm on the fence